axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetLineServiceImpl.java

```diff
@@ -240,6 +262,19 @@ public LinkedHashMap<LocalDate, List<FixedAssetLine>> groupAndSortByDateFixedAss
                 FixedAssetLine::getDepreciationDate, LinkedHashMap::new, Collectors.toList()));
   }
 
+  protected void correctLastFixedAssetLineForProrata(
+      List<FixedAssetLine> fixedAssetLineList,
+      LocalDate firstDepreciationDate,
+      int periodicityInMonth) {
+    int lastIndex = fixedAssetLineList.size() - 1;
+    FixedAssetLine lastFixedAssetLineCopy =
+        fixedAssetLineRepository.copy(fixedAssetLineList.get(lastIndex), false);
+    lastFixedAssetLineCopy.setDepreciationDate(
+        firstDepreciationDate.plusMonths(lastIndex * periodicityInMonth));
+    fixedAssetLineList.remove(lastIndex);
+    fixedAssetLineList.add(lastFixedAssetLineCopy);
```

Comment:
Why don't you just set the depreciationDate in the fixedAssetLineList.get(lastIndex)?